packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -708,6 +728,14 @@ export class PerformanceEvent {
 
 		this.logger.sendPerformanceEvent(event, error);
 	}
+
+	private static readonly eventHits = new Map<string, number>();
+	private static shouldReport(event: ITelemetryGenericEvent, sampleThreshold: number): boolean {
+		const eventKey = `.${event.category}.${event.eventName}`;
+		const hitCount = PerformanceEvent.eventHits.get(eventKey) ?? 0;
+		PerformanceEvent.eventHits.set(eventKey, hitCount + 1);
+		return hitCount % sampleThreshold === 0;
+	}
```

Comment:
Should we reset the hitCount to 0 when we hit the threshold, so a very high-volume event doesn't potentially end up overflowing it? Also, nice clean way to ensure the first event gets emitted, I missed it does it until I looked at the tests 👍.

Reply:
good idea